### ISSUE TITLE
Mypy crash while using sqlalchemy.orm.validates

### ISSUE THREAD

**Issue by zzz4zzz** [sqlalchemy/sqlalchemy#6169]
**Describe the bug**
Using `sqlalchemy.orm.validates` crash mypy check.

**Expected behavior**
Mypy exit with `Success: no issues found in 1 source file`

**To Reproduce**

*test_validates.py*:

```python
from sqlalchemy.orm import validates, declarative_base
from sqlalchemy import Integer, Column, String


Base = declarative_base()


class EmailAddress(Base):
    __tablename__ = 'address'

    id = Column(Integer, primary_key=True)
    email = Column(String)

    @validates('email')
    def validate_email(self, key, address):
        assert '@' in address
        return address
```
*mypy.ini*:
```ini
[mypy]
plugins = sqlalchemy.ext.mypy.plugin
```

**Error**
```
[I] ➜  mypy --show-traceback test_validates.py
test_validates.py:8: error: INTERNAL ERROR -- Please try using mypy master on Github:
https://mypy.rtfd.io/en/latest/common_issues.html#using-a-development-mypy-build
Please report a bug at https://github.com/python/mypy/issues
version: 0.812
Traceback (most recent call last):
  File "mypy/semanal.py", line 4835, in accept
  File "mypy/nodes.py", line 950, in accept
  File "mypy/semanal.py", line 1048, in visit_class_def
  File "mypy/semanal.py", line 1125, in analyze_class
  File "mypy/semanal.py", line 1134, in analyze_class_body_common
  File "mypy/semanal.py", line 1194, in apply_class_plugin_hooks
  File "(...)/test_sqlalchemy/venv/lib/python3.9/site-packages/sqlalchemy/ext/mypy/plugin.py", line 173, in _base_cls_hook
    decl_class._scan_declarative_assignments_and_apply_types(ctx.cls, ctx.api)
  File "(...)/test_sqlalchemy/venv/lib/python3.9/site-packages/sqlalchemy/ext/mypy/decl_class.py", line 125, in _scan_declarative_assignments_and_apply_types
    _scan_declarative_decorator_stmt(cls, api, stmt, cls_metadata)
  File "(...)/test_sqlalchemy/venv/lib/python3.9/site-packages/sqlalchemy/ext/mypy/decl_class.py", line 163, in _scan_declarative_decorator_stmt
    if names._type_id_for_named_node(dec) is names.DECLARED_ATTR:
  File "(...)/test_sqlalchemy/venv/lib/python3.9/site-packages/sqlalchemy/ext/mypy/names.py", line 187, in _type_id_for_named_node
    type_id, fullnames = _lookup.get(node.name, (None, None))
AttributeError: 'CallExpr' object has no attribute 'name'
test_validates.py:8: : note: use --pdb to drop into pdb
```

**Versions.**
 - OS: Ubuntu 20.10
 - Python: Python 3.9.0
 - SQLAlchemy: SQLAlchemy @ git+git://github.com/sqlalchemy/sqlalchemy@42185a1f3866bfc8ebffa2a6c724face8a2db5dc
 - Database: -
 - DBAPI: -

**Comment by zzzeek** [comment]
thanks for reporting.

**Comment by CaselIT** [comment]
Mike should these mypy issues moved to the stub repo, or it's best to keep there only typing issues?

**Comment by zzzeek** [comment]
mypy plugin issues are likely better here but they're going to keep coming in on both sides, no real way to fix that for now

**Comment by zzz4zzz** [comment]
@zzzeek The error is no longer occur. Please close this issue.

**Comment by zzzeek** [comment]
ah ok, I'd want to git bisect and see where this got fixed first.

**Comment by zzzeek** [comment]
fixed in e2008b5541cc155aea538317805e62ff1aa9b300
